cpp/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+
+project(CUDA_DATAFRAME LANGUAGES C CXX CUDA)
+
+###################################################################################################
+# - compiler options ------------------------------------------------------------------------------
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_C_COMPILER $ENV{CC})
+set(CMAKE_CXX_COMPILER $ENV{CXX})
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_CUDA_STANDARD 11)
+set(CMAKE_CUDA_STANDARD_REQURIED ON)
+set(CMAKE_CUDA_SEPARABLE_COMPILATION ON)
+
+if(CMAKE_COMPILER_IS_GNUCXX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+endif(CMAKE_COMPILER_IS_GNUCXX)
+
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_61,code=sm_61")
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_70,code=sm_70")
+
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-extended-lambda --expt-relaxed-constexpr")
+
+# suppress SHFL warnings caused by Modern GPU
+# TODO: remove this when Modern GPU is removed for fixed to use shfl_sync
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Werror cross-execution-space-call -Wno-deprecated-declarations -Xptxas --disable-warnings")
+
+# set warnings as errors
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Werror cross-execution-space-call -Xcompiler -Wall,-Werror")
+
+###################################################################################################
+# - library targets -------------------------------------------------------------------------------
+
+add_library(cudf SHARED
+            src/comm/ipc/ipc.cu
+            src/core/dataframe/column.cpp
```

Comment:
How do you distinguish between "core" and "functions"? Why is "sort" core and "compaction" is not? I think core and functions should be combined here.

Reply:
Flattened directory structure so that there is no longer `core` and `functions`. Just `modules`.